infra/omop_cdm.bicep

```diff
@@ -0,0 +1,192 @@
+targetScope = 'resourceGroup'
+
+param utcValue string = utcNow()
+
+@description('The location for all resources.')
+param location string = resourceGroup().location
+
+@description('The name of webapi CDM database')
+param postgresAtlasDatabaseName string
+
+@description('Used to download the sql scripts from the GitHub repository, this should point to the branch you want to use')
+param branchName string
+
+@description('The name of the postgres server')
+param postgresServerName string
+
+@secure()
+@description('The URL of the container where the CDM data is stored')
+param cdmContainerUrl string
+
+@secure()
+@description('The SAS token to access the CDM data')
+param cdmSasToken string
+
+@secure()
+@description('The name of the OMOP CDM database')
+param postgresOMOPCDMDatabaseName string
+
+@secure()
+@description('Admin password for the postgres server')
+param postgresAdminPassword string
+
+@secure()
+@description('Admin password for the webapi database')
+param postgresWebapiAdminPassword string
+
+@secure()
+@description('Password for the cdm user')
+param postgresOMOPCDMpassword string
+
+@secure()
+@description('The name of the keyvault')
+param keyVaultName string
+
+var postgresOMOPCDMSchemaName = 'cdm'
+var postgresOMOPVocabularySchemaName = 'vocabulary'
+var postgresOMOPResultsSchemaName = 'cdm_results'
+var postgresOMOPTempSchemaName = 'temp'
+var postgresOMOPCDMRole = 'cdm_reader'
+var postgresOMOPCDMUsername = 'cdm_user'
+var postgresOMOPCDMUserSecretName = '${postgresOMOPCDMDatabaseName}-cdm-user-password'
+var postgresOMOPCDMJDBCConnectionStringSecretName = '${postgresOMOPCDMDatabaseName}-cdm-jdbc-connection-string'
+var postgresWebAPISchemaName = 'webapi'
+var postgresWebapiAdminUsername = 'ohdsi_admin_user'
+var postgresAdminUsername = 'postgres_admin'
+var postgresOMOPCDMJDBCConnectionString = 'jdbc:postgresql://${postgresServer.properties.fullyQualifiedDomainName}:5432/${postgresOMOPCDMDatabaseName}?user=${postgresOMOPCDMUsername}&password=${postgresOMOPCDMpassword}&sslmode=require'
+
+// Get the postgres server
+resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' existing = {
+  name: postgresServerName
+}
+
+// Get the keyvault
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
+  name: keyVaultName
+}
+
+// Store OMOM CDM user password in keyvault
+resource postgresAdminSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: postgresOMOPCDMUserSecretName
+  parent: keyVault
+  properties: {
+    value: postgresOMOPCDMpassword
```

Comment:
Should this be here or maybe in the keyvault module? WDYT?

Reply:
Not sure, I think logically those secrets should stay here in this module,
there are also advantages for having this way, 
for instance, if we want to support deployment of another OMOP CDM DB separately and independently with another button? keeping it this way will allow it to be autonomous. WDYT?

Reply:
I agree.